src/main/javascript/modules/admin/commands/test.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2014 Virtue Studios
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * @author Im Frizzy <skype:kfriz1998>
+ * @author Frosty Teh Snowman <skype:travis.mccorkle>
+ * @author Arthur <skype:arthur.behesnilian>
+ * @author Sundays211
+ * @since 05/11/2014
+ */
+import { EventType, Inv } from 'engine/enums';
+import _events from 'engine/events';
+import _entity from 'engine/entity';
+
+import { sendMessage, sendCommandResponse } from 'shared/chat';
+import { requestName } from 'shared/dialog';
+import { lookupPlayerName, getUserHash } from 'shared/util';
+import { runAnim } from 'shared/anim';
+import _coords from 'shared/map/coords';
+import { locationAnim } from 'shared/map/location';
+
+ _events.bindEventListener(EventType.COMMAND_ADMIN, "test", (ctx) => {
+	var args = ctx.cmdArgs;
+	var player = ctx.player;
+	if (args.length < 1) {
+		sendCommandResponse(player, "Usage: test <type>", ctx.console);
+	}
+	switch (args[0]) {
+		case "icons":
+		//	testHeadIcons(player, args, ctx);
+		return;
+		case "name":
+		//	testNameChange(player, args, ctx);
+		return;
+		case "locanim":
+		//	testLocAnim(player, args, ctx);
+		return;
+		case "mestype":
+		//	testMessageType(player, args, ctx);
+		return;
+		case "move":
+		//	testDelayedMovement(player, args, ctx);
+		return;
+	}
+});
+
+//function testHeadIcons(player, args, ctx) {
+//	if (args.length >= 4) {
+		//	player.getHeadIcons().setIcon(parseInt(args[3]), parseInt(args[1]), parseInt(args[2]));
```

Comment:
`getHeadIcons` should also be added. Basically, any `Player` method which isn't already there should be added.

Longer term these will all be replaced by engine methods, but for now we just need their definitions